README.md

```diff
@@ -334,14 +337,19 @@ elements of a list are encoded without an index or key.  They are just encoded
 one after the other, with no need to prefix each element with a key, index
 number nor typ3 byte.
 
-To declare that the List may contain nil elements, the Lists's element typ4
-byte should set the 4th least-significant bit (the "pointer bit") to 1.  If
-(and only if) the pointer bit is 1, each element is prefixed by a 0x00 byte to
-declare that a non-nil item follows, or a 0x01 byte to declare that the next
-item is nil.  Note that the byte values are flipped (typically 0 is used to
-denote nil).  This is to open the possibility of supporting sparse encoding of
-nil lists in the future by encoding the number of nil items to skip as a
-uvarint.
+To declare that the List may contain nil elements (e.g. the list "nillable"),
+the Lists's element typ4 byte should set the 4th least-significant bit (the
+"pointer bit") to 1.  If (and only if) the pointer bit is 1, each element is
+prefixed by a "nil byte" -- a 0x00 byte to declare that a non-nil item follows,
```

Comment:
this seems a bit backwards:
* 0x00 to declare that a non-nil item follows
* 0x01 to declare that a nil item follows

You talk about why you flipped the values in the next sentence but how are we going to encode a uvarint in 1 byte?

Reply:
the nil byte precedes the encoding of the uvarint (or whatever are the elements of the list).

<element typ4 w/ pointer bit set to 1><uvarint(len(elements)=4)><0x01 (nil)><0x01 (nil)><0x00 (not nil)><encoding of 3rd element><0x01 (nil)> as an example with 4 elements.